### PR TITLE
[skip changelog] Update notarization job, now usable with p12 cert format

### PR DESCRIPTION
**Pull request by umbynos**
**Please check if the PR fulfills these requirements**

- [X] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [X] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?**

update the CI
- **What is the current behavior?**

* **What is the new behavior?**

The CI now uses a `.p12` certificates and uses an action to handle keychain unlocking. Other improvements have been added.
Now gon is installed through homebrew.
The `Notarize binary, re-package it and update checksum` step has been split in two different steps: one simple step to notarize and sign and the other one to change permission, calculate the checksum and repackage the binary
- **Does this PR introduce a breaking change?**

no
* **Other information**:

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)
It's necessary to add `INSTALLER_CERT_MAC_P12` and `INSTALLER_CERT_MAC_PASSWORD` as secrets

### CODE REVIEW

**Review by per1234 — CHANGES_REQUESTED** [review]
Thanks Umberto!

I tried it out in my fork with my own Apple Developer ID certificate and it works just fine.

This will also need to be done to the "nightly" workflow:
https://github.com/arduino/arduino-cli/blob/master/.github/workflows/nightly.yaml

File: .github/workflows/release.yaml
Comment (umbynos) [reply .github/workflows/release.yaml:36]: You are right, I might have forgotten about these

File: .github/workflows/release.yaml
Comment (umbynos) [reply .github/workflows/release.yaml:53]: You are right, let's wait for the security review :smile:

File: .github/workflows/release.yaml
Comment (rsora) [inline .github/workflows/release.yaml:55]: @umbynos nice addition! Is it possible to pin the `gon` version?

File: .github/workflows/release.yaml
Comment (rsora) [reply .github/workflows/release.yaml:50]: > I think we can do this import with only a few shell commands. From my notes, I believe these will work:
https://github.com/VSCodium/vscodium/blob/0608a0e812155c63051a07d9fc1489c71273b0ec/sign_mac_app.sh#L9-L18
From the workflow log when I tested this PR, I can see that the Apple-Actions/import-codesign-certs action is running similar commands:

@umbynos I agree with Per on this, please do a doublecheck between the commands orchestrated by the action you proposed and the commands that @per1234 mentioned in the comment. I would like to not have an additional action that orchestrates half a dozen of native commands and also handles our most precious secrets.

**Review by rsora — CHANGES_REQUESTED** [review]
Please check my comments in the code!
(Excellent work BTW!)

File: .github/workflows/release.yaml
Comment (umbynos) [reply .github/workflows/release.yaml:50]: :+1: fix incoming

File: .github/workflows/release.yaml
Comment (umbynos) [reply .github/workflows/release.yaml]: With homebrew taps apparently is not possible, I reverted to the previous installation method

**Review by per1234 — APPROVED** [review]
Looks good!
```
per@Pers-iMac Downloads % spctl -a -vvv -t install "dist/arduino-cli_43.0.0_macOS_64bit/arduino-cli" 
dist/arduino-cli_43.0.0_macOS_64bit/arduino-cli: accepted
source=Notarized Developer ID
origin=Developer ID Application: Per Tillisch (9M5NQMNWBJ)
per@Pers-iMac Downloads % spctl -a -vvv -t install "dist (1)/arduino-cli_0.13.0_macOS_64bit/arduino-cli"
dist (1)/arduino-cli_0.13.0_macOS_64bit/arduino-cli: accepted
source=Notarized Developer ID
origin=Developer ID Application: ARDUINO SA (7KT7ZWMCJT)

```

**Review by rsora — APPROVED** [review]
:+1: 
New secrets configured, let's merge!

**Pull request merged**
